### PR TITLE
feat: improve send modal flow

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This improves two bits in the send flow:

1. Initial flash of "Insufficient funds" error state on max send, caused by `amountCryptoPrecision` not being casted as 0 by bignumber.js in:

https://github.com/shapeshift/web/blob/1eb71c3c109d2c7febfe0a2ef2746f7769ba1b60/src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx#L156

2. Failed XHR errors being wrongly surfaced to the user

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/7084

> What protocols, transaction types or contract interactions might be affected by this PR?

Low

### Network failure message surfaced to the user

1. Let BTC `fees` request fail for a max send (it's currently failing often enough to test this)
2. Ensure the network fetch error isn't surfaced to the user but "Error getting fees" is instead

### Flash of Insufficient Fundus

1. Click max send on BTC
2. Ensure there is no flash of 'Insufficient funds'

### Engineering

- ^

### Operations

- ^

### Error getting fees

image

### No flash of "Insufficient Funds" on max sends

https://jam.dev/c/3470841b-a42d-4ed3-a753-2747767e4399

### CODE REVIEW

**Comment by MBMaria** [comment]
GM - if we can get this into release at some point asap that would be great. tested in develop and fix is working as expected :) @woodenfurniture  @0xApotheosis

**Comment by 0xApotheosis** [comment]
Roger that @MBMaria!

File: src/assets/translations/en/main.json [hunk 1]
```diff
@@ -1044,7 +1044,7 @@
         "tooltip": "Send funds to a single address for THORChain compatibility.",
         "consolidateFunds": "Consolidate"
       },
-      "getBalanceError": "Error getting wallet balances",
+      "getfeesError": "Error getting fees",
```
Comment (0xApotheosis) [inline src/assets/translations/en/main.json:1047]:
```suggestion
      "getFeesError": "Error getting fees",
```

File: src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx [hunk 2]
```diff
@@ -312,6 +312,12 @@ export const useSendDetails = (): UseSendDetailsReturnType => {
     // Since we are debouncing the query, ensure reverting back to an empty input doesn't end up in the previous error being displayed
     if (!hasEnteredPositiveAmount) return setValue(SendFormFields.AmountFieldError, '')
 
+    // openapi-generator error-handling https://github.com/OpenAPITools/openapi-generator/blob/8357cc313be5a099f994c4ffaf56146f40dba911/samples/client/petstore/typescript-fetch/builds/enum/runtime.ts#L221
+    if (
+      error?.message ===
+      'The request failed and the interceptors did not return an alternative response'
+    )
+      return setValue(SendFormFields.AmountFieldError, 'modals.send.getfeesError')
```
Comment (0xApotheosis) [inline src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx:320]:
```suggestion
      return setValue(SendFormFields.AmountFieldError, 'modals.send.getFeesError')
```

**Review by 0xApotheosis — APPROVED** [review]
gm, max send BTC working as expected, plus operations have confirmed against testing steps.

File: src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx (hunk 2 above)
Comment (0xApotheosis) [reply src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx]: https://github.com/shapeshift/web/pull/7095/commits/dfa9b94651f89e21a2e112eb616b3f453a8b9fbd

File: src/assets/translations/en/main.json (hunk 1 above)
Comment (0xApotheosis) [reply src/assets/translations/en/main.json]: https://github.com/shapeshift/web/pull/7095/commits/dfa9b94651f89e21a2e112eb616b3f453a8b9fbd

**Review by 0xApotheosis — COMMENTED** [review]
BTC fees endpoint working correctly after cutover, but sanity checked by forcing the endpoint to fail:

Screenshot 2024-06-12 at 9 41 41 AM

**Pull request merged**